docs/build.md

```diff
@@ -0,0 +1,63 @@
+
+
+# Build
+
+
+The build is organised as follows:
+
+## Top level pipeline jobs
+
+These live at https://ci.adoptopenjdk.net/job/build-scripts/. Currently these jobs are:
+
+* [openjdk8-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/)
+* [openjdk9-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk9-pipeline/)
+* [openjdk10-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk10-pipeline/)
+* [openjdk11-pipeline](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk11-pipeline/)
+
+From he on in this documentation describes the 10 pipeline however it is the same for all Java versions.
+
+The high level sequence of each of these pipelines is shown in:
+
+![build sequence](images/sequence.svg "Build Sequence")
+
+This shows a build of `jdk10-mac-x64-hotspot` and `jdk10-linux-x64-hotspot`. The `jdk10-mac-x64-hotspot` build is expanded as
+an example of how each build pipeline works, however a similar pipeline will be generated for every `jdk10-<os>-<arch>-<impl>`
+that was requested to be built.
+
+### Top level pipeline code
+
+The actions of the pipeline are defined by [openjdk10_pipeline.groovy](/pipelines/build/openjdk10_pipeline.groovy), which 
+calls [BuildBaseFile.groovy](/pipelines/build/BuildBaseFile.groovy).
+
+The actions of this script are:
+
+1. Read in build configuration that determines what configurations have been requested to be built.
+1. For each requested configuration generate a configuration that defines such things as which nodes to run the build on, optional configuration added to [openjdk10_pipeline.groovy](/pipelines/build/openjdk10_pipeline.groovy) etc.
+1. Generate a build job for each requested configuration, these live at [jdk10u](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk10u/) 
+1. Execute all build jobs in parallel
+1. Publish binaries to github
+
+
+#### Generated build jobs
+
+These are generated using the template at [createJobFromTemplate.groovy](/pipelines/build/createJobFromTemplate.groovy)
+This defines a pipeline job that calls [openjdk_build_pipeline.groovy](/pipelines/build/openjdk_build_pipeline.groovy), this script defines the work of what a Build Pipeline does. 
+
+## Build Pipelines
+
+Generated by [createJobFromTemplate.groovy](/pipelines/build/createJobFromTemplate.groovy), and its work defined in [openjdk_build_pipeline.groovy](/pipelines/build/openjdk_build_pipeline.groovy).
```

Comment:
just realised that the file naming std is different, some camel case others underscore separated.